src/Shared/Microsoft.DotNet.Kusto/KustoQuery.cs

```diff
@@ -11,28 +11,50 @@ public class KustoQuery
         public KustoQuery()
         {
             Parameters = new List<KustoParameter>();
+            Options = new Dictionary<string, object>();
         }
 
         public KustoQuery(string text, IEnumerable<KustoParameter> parameters)
         {
             Text = text;
             Parameters = parameters.ToList();
+            Options = new Dictionary<string, object>();
         }
 
-        public KustoQuery(string text) : this(text, new List<KustoParameter>())
+        public KustoQuery(string text, Dictionary<string, object> options)
```

Comment:
It feels a little weird for the options to live on the query, since they are usually about timeouts and other things, that are more to do with the caller and how they want to read the results, rather than "the query".

I'd probably try to wrap the options thing in something more type safe too. The underlying string|string dict might be what it is, but it's a nightmare from an API design, and we can do better with a wrapper. They are already basically undocumented, and now we'll have an API that's taking undocumented options and passing them through as opaque strings, making it even more confusing.

Reply:
If I understood correctly, you are suggesting creating a wrapper (let's call it KustoQueryResquest) that contains a KustoQuery object a dictionary of options?

Reply:
Well, no.  Whatever method takes the query parameter could just take an options object too.  And that object could be a cached static; there is no need to construct a new options object every time... presumably the service will want to make the requests in the same way every time. That's likely why most methods are written the way they are, you have `MakeRequest(Request request, RequestOptions options, CancellationToken token)`.

Then you can just have `private static RequestOptions _theWayIMakeRequestsInThisClass` and when you make a call, `MakeRequest(someRequestIBuilt, _theWayIMakeRequestsInThisClass, token)`, using the same object for every single call.

Reply:
Made a new push, hopefully it looks better

Reply:
@ChadNedzlek, could you please take a look at new changes?